erpnext/education/doctype/program_enrollment/program_enrollment.js

```diff
@@ -57,15 +57,22 @@ frappe.ui.form.on('Program Enrollment', {
 		frm.events.get_courses(frm);
 		if (frm.doc.program) {
 			frappe.call({
-				method: 'erpnext.education.api.get_fee_schedule',
+				method: 'frappe.client.get_list',
 				args: {
-					'program': frm.doc.program,
-					'student_category': frm.doc.student_category
+					doctype: 'Fee Schedule',
+					filters: {
+						'program': frm.doc.program,
+						'student_category': frm.doc.student_category,
+						'academic_year': frm.doc.academic_year
+					},
+					fields: ['academic_term', 'fee_structure', 'student_category', 'due_date', 'total_amount as amount']
 				},
 				callback: function(r) {
 					if (r.message) {
+						cur_frm.clear_table("fees");
+						frm.refresh_fields('fees');
 						frm.set_value('fees' ,r.message);
-						frm.events.get_courses(frm);
+						frm.refresh_fields('fees');
 					}
 				}
 			});
```

Comment:
Can you move this implementation to the server-side to make it API friendly?

https://github.com/frappe/erpnext/wiki/Coding-Standards#all-business-logic-must-be-api-friendly-and-must-be-implemented-in-server-side-python

You can refer `get_program_courses` or `get_students` in `program_enrollment.py` for the implementation.

Reply:
This is specific to program enrollment. So I think this is okay,